.github/workflows/autobuild.yml

```diff
@@ -65,92 +73,190 @@ jobs:
+             uses_codeql:           false
+             
+           - config_name:           MacOS (codeQL)
+             target_os:             macos
+             building_on_os:        macos-10.15
+             cmd1_prebuild:         "sh mac/autobuild/codeQL/autobuild_mac_1_prepare.sh "
+             cmd2_build:            "sh mac/autobuild/codeQL/autobuild_mac_2_build.sh "
+             cmd3_postbuild:        "sh mac/autobuild/codeQL/autobuild_mac_3_copy_files.sh "
+             uses_codeql:           true
+             
+           - config_name:           MacOS (artifacts)
              target_os:             macos
              building_on_os:        macos-10.15
-             asset_path:            deploy/Jamulus-installer-mac.dmg
-             asset_name:            jamulus_${{ needs.create_release.outputs.version_name }}_mac.dmg
-           - config_name:           Windows
+             cmd1_prebuild:         "sh mac/autobuild/artifacts/autobuild_mac_1_prepare.sh "
+             cmd2_build:            "sh mac/autobuild/artifacts/autobuild_mac_2_build.sh "
+             cmd3_postbuild:        "sh mac/autobuild/artifacts/autobuild_mac_3_copy_files.sh "
+             uses_codeql:           false
+             
+           - config_name:           Windows (artifact+codeql)
              target_os:             windows
              building_on_os:        windows-latest
-             asset_path:            deploy\Jamulus-installer-win.exe
-             asset_name:            jamulus_${{ needs.create_release.outputs.version_name }}_win.exe
-           - config_name:           AndroidAPK
-             target_os:             android
-             building_on_os:        ubuntu-18.04
-             asset_path:            android-build/build/outputs/apk/debug/android-build-debug.apk
-             asset_name:            jamulus_${{ needs.create_release.outputs.version_name }}_android.apk
+             cmd1_prebuild:         powershell ./windows/autobuild/autobuild_windowsinstaller_1_prepare.ps1
+             cmd2_build:            powershell ./windows/autobuild/autobuild_windowsinstaller_2_build.ps1
+             cmd3_postbuild:        powershell ./windows/autobuild/autobuild_windowsinstaller_3_copy_files.ps1
+             uses_codeql:           true 
+         
+    runs-on:                        ${{ matrix.config.building_on_os }}
     steps:
-
       # Checkout code
       - name:                       Checkout code
         uses:                       actions/checkout@v2
         with:
           submodules:               true
+          
+      # Prepare (install QT & dependencies)
+      - name:                       "Prepare for ${{ matrix.config.config_name }}"
+        if:                         ${{ matrix.config.cmd1_prebuild }}
+        run:                        ${{ matrix.config.cmd1_prebuild }}
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'  # allow setting environment variables
```

Comment:
Sounds a bit dangerous. Can't you just output environment variables via echo?

Reply:
this allos the use of ::set-env name=xxxxxx::${xxxxxxx}

My problem was that autobuild_apk_1_prepare.sh sets a lot of variables, and I dont know which ones are necessary for building then. Doint special treatment for 15 variables via output and mentioning them as environment of next step is not clean any more. I'm open for suggestions.

Reply:
Hmm. If you think this is the best way to do it, I would just leave it like that for now.